src/plugins/data_view_field_editor/public/components/preview/field_list/field_list.tsx

```diff
@@ -55,20 +55,15 @@ export const PreviewFieldList: React.FC<Props> = ({ height, clearSearch, searchV
 
   const [showAllFields, setShowAllFields] = useState(false);
 
-  const {
-    fields: { getAll: getAllFields },
-  } = dataView;
-
-  const indexPatternFields = useMemo(() => {
-    return getAllFields();
-  }, [getAllFields]);
-
   const fieldList: DocumentField[] = useMemo(
     () =>
-      indexPatternFields
-        .map(({ name, displayName }) => {
+      dataView.fields
+        .getAll()
+        .map((field) => {
+          const { name, displayName } = field;
+          const formatter = dataView.getFormatterForField(field);
           const value = get(currentDocument?._source, name);
-          const formattedValue = defaultValueFormatter(value);
+          const formattedValue = formatter.convert(value, 'html') ?? defaultValueFormatter(value);
```

Comment:
There's no need to look up the default formatter as `getFormatterForField` includes that functionality.